[COSMETIC]
Workaround anoying xmms behaviour. During a title change,
xmms_remote_get_volume() returns -1, which gets the slider displayed.

--- src/xmms_plugin/xmms_osd.c.orig	2003-09-26 02:26:17.000000000 +0200
+++ src/xmms_plugin/xmms_osd.c	2003-09-28 23:01:46.000000000 +0200
@@ -430,11 +430,17 @@ timeout_func (gpointer data)
     }
   else if (current.volume != previous.volume && show.volume)
     {
+      DEBUG ("V: %d->%d\n", previous.volume, current.volume);
+      /* xmms returns -1 during a title change. skip this and try again later. */
+      if ((previous.volume == -1) || (current.volume == -1))
+	goto skip;
       xosd_display (osd, 0, XOSD_string, "Volume");
       xosd_display (osd, 1, XOSD_percentage, current.volume);
     }
   else if (current.balance != previous.balance && show.balance)
     {
+      DEBUG ("B: %d->%d\n", previous.balance, current.balance);
+      /* FIXME: Same as above might happen, but with what values? */
       xosd_display (osd, 0, XOSD_string, "Balance");
       xosd_display (osd, 1, XOSD_slider, current.balance);
     }
@@ -449,6 +455,7 @@ timeout_func (gpointer data)
       xosd_display (osd, 1, XOSD_string, current.shuffle ? "On" : "Off");
     }
 
+skip:
   if (title2)
     free (title2);
 
